src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -382,13 +382,21 @@ static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
 *
 * emptyAttributeValue -
 *
-* Check that the attribute doesn't have any value, taking into account its multi-valuated
-* nature
+* Check that the attribute doesn't have any value, taking into account its multi-valuated nature.
+*
+* NOTE
+*   1. ValueTypeNumber and ValueTypeBoolean are always non-empty.
+*   2. In API version 2, the value of a ContextAttribute CANNOT be left out
```

Comment:
As discussed over skype. This behavior will change soon, so a FIXME has been added.
Fixed in 2553d84.